src/fu-bluez-backend.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (C) 2021 Ricardo Cañuelo <ricardo.canuelo@collabora.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuBackend"
+
+#include "config.h"
+
+#include "fu-ble-device.h"
+#include "fu-bluez-backend.h"
+
+#define DEFAULT_PROXY_TIMEOUT	5000
+
+struct _FuBluezBackend {
+	FuBackend		 parent_instance;
+	GDBusConnection		*connection;
+	GHashTable		*devices;	/* platform_id : * FuDevice */
+};
+
+G_DEFINE_TYPE (FuBluezBackend, fu_bluez_backend, FU_TYPE_BACKEND)
+
+
+/*
+ * Returns a new FuBleDevice populated with the properties passed
+ * in the properties variant (@a{sv}).
+ *
+ * TODO: Search device characteristics and populate the hash table.
+ */
+static FuBleDevice *
+fu_bluez_load_device_properties (GVariant *properties)
+{
+	const gchar *prop_name;
+	GVariantIter it;
+	g_autoptr(GVariant) prop_val = NULL;
+	FuBleDevice *dev = NULL;
+	g_autofree gchar *address = NULL;
+	g_autofree gchar *name = NULL;
+	gboolean connected = FALSE;
+	GHashTable *characteristics = g_hash_table_new_full (g_str_hash, g_str_equal,
+							     g_free, g_free);
+
+	g_variant_iter_init (&it, properties);
+	while (g_variant_iter_next (&it, "{&sv}", &prop_name, &prop_val)) {
+		if (g_strcmp0 (prop_name, "Address") == 0) {
+			address = g_strdup (g_variant_get_string(prop_val, NULL));
+			continue;
+		}
+		if (g_strcmp0 (prop_name, "Name") == 0) {
+			name = g_strdup (g_variant_get_string(prop_val, NULL));
+			continue;
+		}
+		if (g_strcmp0 (prop_name, "Connected") == 0) {
+			connected = g_variant_get_boolean(prop_val);
+			continue;
+		}
```

Comment:
did I miss it - how does this get updated?

Reply:
There's a big fat:

	/*
	 * TODO: Subscribe DBus signals.
	 */

:)

Reply:
Fixed.